qemu/tests/x86_cpu_flags.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from avocado.utils import cpu
+from virttest import error_context, env_process
+from qemu.tests.x86_cpu_model import check_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flags.
+    1) Check if current flags are in the supported lists, if no, cancel test
+    2) Otherwise, boot guest with the cpu flags
+    3) Check cpu flags in guest(only for linux guest)
+    4) Reboot guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    vendor_id = params.get("vendor_id", "")
+    if vendor_id:
+        if vendor_id != cpu.get_vendor():
+            test.cancel("This host vendor doesn't match the %s" % vendor_id)
+
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_flags(params, flags, test)
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+    if params["os_type"] == "linux":
+        error_context.context("Check cpu flags inside guest", logging.info)
+        check_flags(params, flags, test, session)
+
+    if params.get("reboot_method"):
+        error_context.context("Reboot guest '%s'." % vm.name, logging.info)
+        vm.reboot(session=session)
```

Comment:
session = vm.reboot(session=session)?

Reply:
> session = vm.reboot(session=session)?

seems `vm.reboot` needs a session when the `method` is 'shell' and if we don't give it a session it will get one by itself.
```
def reboot(self, session=None, method="shell", nic_index=0,
               timeout=virt_vm.BaseVM.REBOOT_TIMEOUT, serial=False):
       ............
        def _shell_reboot(session, timeout):
            if not session:
                if not serial:
                    session = self.wait_for_login(nic_index=nic_index,
                                                  timeout=timeout)
                else:
                    session = self.wait_for_serial_login(timeout=timeout)
            reboot_cmd = self.params.get("reboot_command")
            logging.debug("Send command: %s" % reboot_cmd)
            session.cmd(reboot_cmd, ignore_all_errors=True)
```

Reply:
@nanliu-r Yes I understand, my point is, you have `session.close()`, so here your session should change to the one `reboot` return to you, the point is `session = vm.reb...`

Reply:
@vivianQizhu got it and update, thanks.